components/sync15/src/error.rs

```diff
@@ -112,14 +106,16 @@ pub enum ErrorKind {
     #[fail(display = "Unexpected server behavior during batch upload: {}", _0)]
     ServerBatchProblem(&'static str),
 
-    #[fail(display = "Setup state machine cycle detected")]
-    SetupStateCycleError,
+    #[fail(
+        display = "It appears some other client is also trying to setup storage; try again later"
+    )]
+    SetupRace,
 
     #[fail(display = "Client upgrade required; server storage version too new")]
     ClientUpgradeRequired,
 
-    #[fail(display = "Setup state machine disallowed state {}", _0)]
-    DisallowedStateError(&'static str),
+    #[fail(display = "Our storage needs setting up and we can't do it")]
```

Comment:
Maybe make it clear (in a comment, even) that `allowed_states` is what forbids this.